go-controller/pkg/ovn/egressip.go

```diff
@@ -2099,11 +2101,31 @@ func (oc *Controller) checkEgressNodesReachability() {
 }
 
 func (oc *Controller) isReachable(node *egressNode) bool {
-	for _, ip := range node.mgmtIPs {
-		if dialer.dial(ip) {
-			return true
+	var maxRetryCnt int
+	var retryTimeOut time.Duration
+
+	switch oc.eIPC.egressIPNodeReachConfig.EgressIPReachabiltyTimeout {
+	// Check if we need to do node reachability check
+	case 0:
+		klog.V(5).Infof("Skip EgressIP node reachability check for node %s", node.name)
+		return true
+	case 1:
+		maxRetryCnt = 3
+		retryTimeOut = 333 * time.Millisecond
+	default:
+		maxRetryCnt = oc.eIPC.egressIPNodeReachConfig.EgressIPReachabiltyTimeout
+		retryTimeOut = 1 * time.Second
+	}
+
+	for retryCnt := 0; retryCnt < maxRetryCnt; retryCnt++ {
+		for _, ip := range node.mgmtIPs {
+			if dialer.dial(ip, retryTimeOut) {
+				return true
+			}
 		}
+		time.Sleep(retryTimeOut)
```

Comment:
> What about using a ticker? https://pkg.go.dev/time#NewTicker Because wouldn't dialer.dial(..., retryTimeOut) and time.Sleep(retryTimeOut) in the worst case compound to 2x retryTimeOut?

idk about using ticker from reading the doc doesn't seem right option for this but not sure @trozet WDYT ?

Reply:
I thought of something like this (perhaps a bit risky with the unbuffered channel, might want to make it buffered):
https://go.dev/play/p/z5CehNGGGpS?v=goprev

If you run this in the playground, then the dial.dialer returns immediately (the min() case). If you run the same test on your computer, it will take the maximum time (the max() case).

Ironically, the first tick is always offset by `retryTimeout`, meaning hat successful tests are offset by a sec. 

Perhaps that's overkill, but the current implementation of isReachable has a variable runtime.

Reply:
the only reason Ticker version finish on the configured total timeout is because the context.WithTimeOut setting , the strange things here is `DialTimeout()` returns immediatly as if I am calling `Dial()` timeout value is completely ignored that is why reachable function w/o Ticker just func of sleep delay, I will need to investigate this more not sure if its the playground issue as this is how other users of DialTimeout use it too for example etcd3

Reply:
with 5s config it takes `5.505563053s` with single mgmt interface and `10.507464826s` my understanding of this knob is totalTimeout/ mgmt intf , anyway I think this good/simple enough for this operation w/o need for additional complexity